src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -22070,15 +22070,15 @@ struct S
 }";
             var comp = CreateCompilation(new[] { source, UnscopedRefAttributeDefinition });
             comp.VerifyDiagnostics(
-                // (4,6): error CS9063: UnscopedRefAttribute cannot be applied to this item because it is unscoped by default.
+                // (4,6): error CS9101: UnscopedRefAttribute can only be applied to struct instance methods, and cannot be applied to constructors or init-only methods.
```

Comment:
>cannot be applied to constructors [](http://example.com/codeflow?start=113&length=33)

nit: The diagnostic says "cannot be applied to constructors" but it looks like we don't report this diagnostic on constructors (see `UnscopedRefAttribute_Constructor`) so it doesn't seem particularly relevant. Consider removing this portion of the diagnostic.

Reply:
Good catch, thanks. I will leave the diagnostic message as is since the message is correct that the attribute cannot be applied to constructors. However, we typically report `ErrorCode.ERR_AttributeOnBadSymbolType` instead of this error because the BCL attribute definition does not include `AttributeTargets.Constructor`. To ensure we are catching this case, regardless of the attribute definition, I've added a version of the constructor test with an attribute that does allow targeting constructors.